cmd/infra/powervs/create.go

```diff
@@ -619,102 +610,95 @@ func (infra *Infra) createCloudInstance(options *CreateInfraOptions) (resourceIn
 
 	infra.Stats.CloudInstance.Duration.Duration = time.Since(startTime)
 
-	return
+	return resourceInstance, nil
 }
 
 // getAccount getting the account id from core.Authenticator
-func getAccount(auth core.Authenticator) (accountID string, err error) {
+func getAccount(auth core.Authenticator) (string, error) {
 	iamv1, err := iamidentityv1.NewIamIdentityV1(&iamidentityv1.IamIdentityV1Options{
 		Authenticator: auth,
 		URL:           getCustomEndpointUrl(platformService, iamidentityv1.DefaultServiceURL),
 	})
 	if err != nil {
-		return
+		return "", err
 	}
 
 	apiKeyDetailsOpt := iamidentityv1.GetAPIKeysDetailsOptions{IamAPIKey: &cloudApiKey}
 	apiKey, _, err := iamv1.GetAPIKeysDetails(&apiKeyDetailsOpt)
 	if err != nil {
-		return
+		return "", err
 	}
 	if apiKey == nil {
-		err = fmt.Errorf("could not retrieve account id")
-		return
+		return "", fmt.Errorf("could not retrieve account id")
 	}
 
-	accountID = *apiKey.AccountID
-	return
+	return *apiKey.AccountID, nil
 }
 
 // createPowerVSSession creates PowerVSSession of type *ibmpisession.IBMPISession
-func createPowerVSSession(accountID string, powerVSRegion string, powerVSZone string, debug bool) (session *ibmpisession.IBMPISession, err error) {
+func createPowerVSSession(accountID string, powerVSRegion string, powerVSZone string, debug bool) (*ibmpisession.IBMPISession, error) {
 	auth := getIAMAuth()
 
-	if err != nil {
-		return
-	}
-
 	opt := &ibmpisession.IBMPIOptions{Authenticator: auth,
 		Debug:       debug,
 		URL:         getCustomEndpointUrl(powerVsService, powerVsDefaultUrl(powerVSRegion)),
 		UserAccount: accountID,
 		Zone:        powerVSZone}
 
-	session, err = ibmpisession.NewIBMPISession(opt)
-	return
+	session, err := ibmpisession.NewIBMPISession(opt)
+	return session, err
 }
 
 // createVpcService creates VpcService of type *vpcv1.VpcV1
-func createVpcService(region string, infraID string) (v1 *vpcv1.VpcV1, err error) {
-	v1, err = vpcv1.NewVpcV1(&vpcv1.VpcV1Options{
+func createVpcService(region string, infraID string) (*vpcv1.VpcV1, error) {
+	v1, err := vpcv1.NewVpcV1(&vpcv1.VpcV1Options{
 		ServiceName:   "vpcs",
 		Authenticator: getIAMAuth(),
 		URL:           getCustomEndpointUrl(vpcService, vpcDefaultUrl(region)),
 	})
 	log(infraID).Info("Created VPC Service for", "URL", v1.GetServiceURL())
-	return
+	return v1, err
 }
 
 // setupPowerVSCloudInstance takes care of setting up powervs cloud instance
-func (infra *Infra) setupPowerVSCloudInstance(options *CreateInfraOptions) (err error) {
+func (infra *Infra) setupPowerVSCloudInstance(options *CreateInfraOptions) error {
 	log(options.InfraID).Info("Setting up PowerVS Cloud Instance ...")
 	var cloudInstance *resourcecontrollerv2.ResourceInstance
-	if options.PowerVSCloudInstanceID != "" {
-		log(options.InfraID).Info("Validating PowerVS Cloud Instance", "id", options.PowerVSCloudInstanceID)
-		cloudInstance, err = validateCloudInstanceByID(options.PowerVSCloudInstanceID)
+	var err error
```

Comment:
do we really need this err variable outside the if/else block? I don't its reference after this block